.pre-commit-config.yaml

```diff
@@ -0,0 +1,30 @@
+minimum_pre_commit_version: '2.17'
+repos:
+  # shell scripts
+  - repo: https://github.com/detailyang/pre-commit-shell
+    rev: 1.0.5
+    hooks:
+      - id: shell-lint
+        args: [ --format=json ]
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.1.0
+    hooks:
+      - id: check-added-large-files
+      - id: check-docstring-first
+      - id: check-merge-conflict
+      - id: check-shebang-scripts-are-executable
+      - id: detect-private-key
```

Comment:
This feels very ambitious as even a little string like a github token could be a security issue. I would leave this out.

Reply:
IIRC we have had cases of creds getting pushed to repos but not sure if they were keys. 

It can't catch all cases but IMO it's still useful. But I'm fine with removing if we don't think its needed.

Reply:
I would leave it. It's a simple text match to check for common formats of API keys. It's really pretty fast, so it won't hurt to leave it, IMHO.

Reply:
Fine with leaving it